templates/freeradius/sql.j2

```diff
@@ -0,0 +1,41 @@
+sql {
+  driver = "{{ freeradius_sql.driver }}"
+  dialect = "{{ freeradius_sql.dialect }}"
+{% if freeradius_sql.dialect in ['postgresql', 'mysql'] %}
+	radius_db = "{{ openwisp2_database.name }}"
+  host = "{{ openwisp2_database.host }}"
+  port = "{{ openwisp2_database.port }}"
+  user = "{{ openwisp2_database.user }}"
+  password = "{{ openwisp2_database.password }}"
+{% elif 'sqlite' == freeradius_sql.dialect %}
+	sqlite {
+		filename = "{{ openwisp2_database.name }}"
+	}
```

Comment:
I am getting:
```
rlm_sql_sqlite: libsqlite version: 3.27.2
   sqlite {
        filename = "/opt/openwisp2/db.sqlite3"
        busy_timeout = 200
   }
rlm_sql_sqlite: Database exists, but couldn't be opened: Permission denied
/etc/freeradius/3.0/mods-enabled/sql[1]: Instantiation failed for module "sql"
```

on this line.
I don't understand, it was working last time I checked. 
Was some setting for this changed recently? or something I am doing incorrectly? What about the version is `3.27.2` used by you as well? @nemesisdesign

Reply:
look for the task `set permissions to sqlite db`, the sqlite DB by default is readable only by the user www-data and the group www-data.

A possible solution may be to add the freeradius user to the www-data group.

Reply:
I think I made the db have 777 permissions when I was last testing, but I'll check again.
From what I remember the problem was because only one application can write sqlite db at a time and freeradius and the other one were fighting for the write permission.

However, I'll double check this.

Reply:
I think is simply a permission issue, the sqlite DB is readable only by group and owner, the freeradius user is not owner nor group.

Yes, only one process can write to the DB at time, if more processes try to write while another process is writing they'll get an error, but that won't stop the other processes from reading the DB (the celery worker for example reads the DB while the django application also reads it, so freeradius can read it too).

The write operations are performed via the openwisp-radius API, hence only django will write to the DB, freeradius should not be writing to the DB directly unless I'm forgetting anything.

Reply:
It's working! :smile: